### PR TITLE
Add callback function for poll events to client api

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
This PR addresses the issue raised, https://answers.launchpad.net/libdrizzle/+question/233212, 
in which the removal of `event_watch_fn` meant that a client could no longer 
suscribe to and get notified about poll events

### CODE REVIEW

**Comment by mathias-baumann-sociomantic** [comment]
LGTM

**Comment by ben-palmer-sociomantic** [comment]
Generally LGTM but perhaps it would be best to add an entry to the release notes about this change.

**Comment by andreas-bok-sociomantic** [comment]
Agree on https://github.com/sociomantic/libdrizzle-redux/pull/5#issuecomment-162860691 
though should the first "release" simply be the officieal `libdrizzle redux v5.1.4`

**Comment by leandro-lucarella-sociomantic** [comment]
You can assume the official v5.1.4 was already released and call next release v5.2.0 for example, so the release notes includes all changes between the official v5.1.4 and v.5.2.0.

**Comment by ben-palmer-sociomantic** [comment]
ping @andreas-bok-sociomantic once this has the release notes I am happy to rebase it.

**Comment by andreas-bok-sociomantic** [comment]
@ben-palmer-sociomantic, added release notes.

**Comment by andreas-bok-sociomantic** [comment]
Added documentation for `drizzle_set_event_watch_fn` to the libdrizzle manual

File: docs/api/connection.rst [hunk 1]
```diff
@@ -194,7 +203,7 @@ Functions
 
    :param options: The options object to get the value from
    :returns: The state of the auth plugin option
-   
+
```
Comment (ben-palmer-sociomantic) [inline docs/api/connection.rst]: This whitespace change should be in its own commit.

**Comment by ben-palmer-sociomantic** [comment]
Apart from my comment about the whitespace seems to be looking good, can you squash together any commits that need it before I rebase?

**Comment by andreas-bok-sociomantic** [comment]
Squashed commits. Left whitespace fix out of this commit to keep the history more clean.

**Comment by ben-palmer-sociomantic** [comment]
LGTM, @mathias-baumann-sociomantic any other comments?

**Comment by mathias-baumann-sociomantic** [comment]
No comments.

**Comment by ben-palmer-sociomantic** [comment]
This pull request has been rebased via `git hub pull rebase`. Original pull request HEAD was 7ad230ede5dab4b56fe066be384a966a28553447, new (rebased) HEAD is 0034b383f72f15c340c94a8b6ed8e1ebd0d12715
